MdeModulePkg/Core/Dxe/Misc/MemoryProtection.c

```diff
@@ -1554,6 +1554,18 @@ ApplyMemoryProtectionPolicy (
   //   // newly added region of a type that does not require protection
   //   return EFI_SUCCESS;
   // }
+
+  // To catch the edge case where the attributes are not consistent across the range, get the
+  // attributes from the page table to see if they are consistent. If they are not consistent,
+  // GetMemoryAttributes() will return an error.
+  if (MemoryAttributeProtocol != NULL) {

```

Comment:
nit:
MemoryAttributeProtocol 

This is a global variable defined in ? Should it be mMemoryAttributeProtocol ?

Reply:
It's declared in DxeMain.h, and I think that means it should have the 'g' prefix convention. I can post a separate PR to change all references to gMemoryAttributeProtocol if that sounds right.